app/javascript/src/ActiveDocs/components/ActiveDocsSpec.jsx

```diff
@@ -4,15 +4,84 @@ import React from 'react'
 
 import { createReactWrapper } from 'utilities/createReactWrapper'
 import SwaggerUI from 'swagger-ui-react'
-
 import 'swagger-ui-react/swagger-ui.css'
 
 type ActiveDocsSpecProps = {
-  url: string
+  accountData: {
+    [string]: Array<mixed>
+  },
+  url: string,
+}
+
+const AUTOCOMPLETE_PARAMS_MAP = {
+  'app_key': 'app_keys',
+  'app_id': 'app_ids',
+  'application_id': 'application_ids',
+  'user_key': 'user_keys',
+  'user_id': 'user_ids',
+  'account_id': 'account_ids',
+  'metric_name': 'metric_names',
+  'metric_id': 'metric_ids',
+  'backend_api_metric_name': 'backend_api_metric_names',
+  'service_id': 'service_ids',
+  'admin_id': 'admin_ids',
+  'service_plan_id': 'service_plan_ids',
+  'application_plan_id': 'application_plan_ids',
+  'account_plan_id': 'account_plan_ids',
+  'client_id': 'client_ids',
+  'client_secret': 'client_secrets',
+  'service_token': 'service_tokens',
+  'access_token': 'access_token'
+}
+
+const getParam = (param, accountData) => {
+  const paramName = AUTOCOMPLETE_PARAMS_MAP[param.name]
+  const paramHasAutocompleteData = (accountData[paramName].length > 0) &&
+    accountData[paramName].every(param => param.value !== '')
+  if (!paramHasAutocompleteData) {
+    return param
+  }
+  const autocompleteData = accountData[paramName]
+  const examples = { '0': { summary: 'Select an option to autocomplete', value: '-' } }
+  autocompleteData.forEach((item, index) => {
+    examples[`${index + 1}`] = {
```

Comment:
Q: does it have to be a string?
```suggestion
    examples[index + 1] = {
```

Reply:
That's because this old flow issue: https://github.com/facebook/flow/issues/380
`Non-string literal property keys not supported.`